test/test_torch.py

```diff
@@ -12151,6 +12151,25 @@ def test_ceil_out_mismatch(self, device):
         b = torch.randn(1, device=device)
         self.assertRaises(RuntimeError, lambda: torch.ceil(a, out=b))
 
+    # Test for unary ops appropriate for floating-type output from integral and boolean inputs
+    @dtypes(torch.int8, torch.int16, torch.int32, torch.int64, torch.bool)
+    def test_intbool_to_float_upcasting(self, device, dtype):
```

Comment:
should this go in test_type_promotion.py?

Reply:
Sure! I'll move it to test_type_promotion.py

Reply:
There is a plan to create a Test Class for all the Unary Ufuncs, as per discussion with @mruberry. This test was although supposed to go in `test_type_promotion.py` as correctly suggested by @nairbv, but since we are generalizing it for all the unary ufuncs, so we'll keep it in `test_torch.py`.